frontend/src/metabase/query_builder/containers/QueryBuilder.jsx

```diff
@@ -423,17 +424,40 @@ function QueryBuilder(props) {
 
   const isLocationAllowed = useCallback(
     location => {
-      if (!question?.isNative() || question?.isDataset()) {
+      const isNewQuestion = !originalQuestion;
```

Comment:
Could you move this to a utility function and test it?

Reply:
8d8c33eca6dda7931df9ce2f410b7ffe57f50caf...419c094a6df44e46d358f41c75911c9c7c44bf66